stdlib/LinearAlgebra/test/diagonal.jl

```diff
@@ -1266,4 +1266,13 @@ end
     @test copy(Diagonal(1:5)) === Diagonal(1:5)
 end
 
+@testset "kron! for Diagonal" begin
+    a = Diagonal([2,2])
+    b = Diagonal([1,1])
+    c = Diagonal([0,0,0,0])
+    @test kron!(c,b,a) == Diagonal([2,2,2,2])
```

Comment:
```suggestion
    kron!(c,b,a)
    @test c  == Diagonal([2,2,2,2])
```
Similarly, for the other one. This tests that `c` is being overwritten with the result.